### DISCUSSION TITLE
Lightweight Speech-to-Text transcriber

### DISCUSSION THREAD

**Discussion by bobobubs** [openai/whisper#1744] [Show and tell]
Hello everyone,

I've recently developed a basic python program that allows for seamless audio recording and transcription using OpenAI's Whisper model. The script is designed to trigger audio recording with a simple hotkey press, save the recorded audio as a WAV file, and transcribe the audio to text. The transcription is then automatically typed out wherever your cursor is located.
Features

-     Hotkey Activated Recording: Start and stop recording with a hotkey press (Alt+X by default).
-     Real-time Transcription: As soon as the recording stops, the audio is transcribed using the Whisper model.
-     Automatic Typing: The transcribed text is typed out automatically wherever your cursor is.

https://github.com/bobobubs/whisper-speech-to-text

**Comment by Purfview** [comment]
Sounds interesting. Is this cross-platform compatible?

**Reply by bobobubs** [reply]
The README has directions for setting it up on windows. BUT, the only requirements are that the user have the correct Python packages and ffmpeg. If you get those setup on macOS or Linux it ideally should still work. There is a flag that allows you to set the hotkey to something more appropriate for whatever platform you want to use.
